### ISSUE TITLE
[continuumio/anaconda3:latest] TensorFlow package installation consumes much more time now

### ISSUE THREAD

**Issue by alexander-smolyakov** [ContinuumIO/docker-images#389]
## Description

With the latest release of the `continuumio/anaconda3` Docker container, we noticed that installation packages like `TensorFlow` required much more time (Up to several hours) than in the `continuumio/anaconda3:2023.03-1`.

It seems conda tries to resolve some conflicts in the environment during installation:

```console
conda install tensorflow
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: ...working... unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working...
....
```

For comparison, log from the previous version of the Docker container:

```console
conda install tensorflow                             
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done              
                                                     
## Package Plan ##                                   
                                                     
  environment location: /opt/conda                   
                                                     
  added / updated specs:                             
    - tensorflow                                     
                                                    
```

We have tested the latest version of the `miniconda` Docker container, and it works fine compared to the latest version of the `anaconda` container. It got me thinking that issue could be related not to the latest version of the conda but to the `anaconda` container itself:

``` console
conda install tensorflow
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /opt/conda

  added / updated specs:
    - tensorflow
```

## Repro sample

```Dockerfile
FROM continuumio/anaconda3

RUN \
  echo "conda update conda" \
  && conda update -y conda \
  && echo "conda install tensorflow" \
  && conda install -c conda-forge --yes tensorflow \
  && echo "conda install pytorch" \
  && conda install -c conda-forge --yes pytorch
```

## Other samples used for testing

```Dockerfile
FROM continuumio/miniconda3

RUN \
  echo "conda update conda" \
  && conda update -y conda \
  && echo "conda install tensorflow" \
  && conda install -c conda-forge --yes tensorflow \
  && echo "conda install pytorch" \
  && conda install -c conda-forge --yes pytorch
```

```Dockerfile
FROM continuumio/anaconda3:2023.03-1

RUN \
  echo "conda update conda" \
  && conda update -y conda \
  && echo "conda install tensorflow" \
  && conda install -c conda-forge --yes tensorflow \
  && echo "conda install pytorch" \
  && conda install -c conda-forge --yes pytorch

```

**Comment by dbast** [comment]
@alexander-smolyakov you are mixing different channels here:
* the anaconda3 container has all the anaconda3 packages from defaults installed into the so called base env 
* the miniconda3 container has only conda + its dependencies in the base env
* you try to install tensorflow / pytorch from *conda-forge*, which is an entire different channel compared to defaults!
* it is best practise to leave the base env either untouched (and instead create a new env with the packages from a different channel) OR install tensorflow / pytorch from defaults.... reason is that installing packages from different channels with many dependencies easily causes conflicts.... the more there already is in the base env (lots of stuff in the anaconda3 case) the more likely this happens.

in other words:
```Dockerfile
FROM continuumio/miniconda3

# lets create a new env
RUN conda create --name test-env -c conda-forge --yes \
    tensorflow \
    pytorch
```
OR
```Dockerfile
FROM continuumio/anaconda3

# lets install into base, but only packages from defaults (no extra --channel parameter)
RUN conda install --yes \
    tensorflow \
    pytorch
```

**Comment by alexander-smolyakov** [comment]
@dbast, thanks for your reply. I tested the suggested options. Creating a new environment works fine.

> OR install tensorflow / pytorch from defaults....

```Dockerfile
FROM continuumio/anaconda3

# lets install into base, but only packages from defaults (no extra --channel parameter)
RUN conda install --yes \
    tensorflow \
    pytorch
```

When trying installation from the default channel, the same issue appears, and conda tries to resolve some conflicts in the environment during installation which take a lot of time:

```Console
=> [2/2] RUN   echo "conda install tensorflow"   && conda install --yes tensorflow   && echo "conda install pytorch"   && conda install --yes pytorch             676.8s
 => => # Collecting package metadata (current_repodata.json): ...working... done
 => => # Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
 => => # Solving environment: ...working... unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.
 => => # Collecting package metadata (repodata.json): ...working... done
 => => # Solving environment: ...working... unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
 => => # Solving environment: ...working...
```

I wander if something is broken in the base environment.
